Content.Client/Chat/ChatBox.cs

```diff
@@ -1,29 +1,67 @@
-﻿using Content.Shared.Chat;
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Content.Client.State;
+using Content.Client.UserInterface;
+using Content.Client.UserInterface.Stylesheets;
+using Content.Client.Utility;
+using Content.Shared.Chat;
 using Robust.Client.Graphics;
+using Robust.Client.ResourceManagement;
+using Robust.Client.State;
 using Robust.Client.UserInterface;
 using Robust.Client.UserInterface.Controls;
 using Robust.Shared.Input;
+using Robust.Shared.IoC;
 using Robust.Shared.Localization;
 using Robust.Shared.Maths;
+using Robust.Shared.Timing;
 using Robust.Shared.Utility;
 
 namespace Content.Client.Chat
 {
     public class ChatBox : Control
     {
+        public const float InitialChatBottom = 235;
+
         public delegate void TextSubmitHandler(ChatBox chatBox, string text);
 
-        public delegate void FilterToggledHandler(ChatBox chatBox, BaseButton.ButtonToggledEventArgs e);
+        public delegate void FilterToggledHandler(ChatChannel toggled, bool enabled);
+
+        public event TextSubmitHandler? TextSubmitted;
+
+        public event FilterToggledHandler? FilterToggled;
 
         public HistoryLineEdit Input { get; private set; }
         public OutputPanel Contents { get; }
 
-        // Buttons for filtering
-        public Button AllButton { get; }
-        public Button LocalButton { get; }
-        public Button OOCButton { get; }
-        public Button AdminButton { get; }
-        public Button DeadButton { get; }
+        public event Action<ChatResizedEventArgs>? OnResized;
+
+        // order in which the available channel filters show up when available
+        public static readonly IReadOnlyList<ChatChannel> ChannelFilterOrder = new List<ChatChannel>
+        {
+            ChatChannel.Local, ChatChannel.Emotes, ChatChannel.Radio, ChatChannel.OOC, ChatChannel.Dead, ChatChannel.AdminChat,
+            ChatChannel.Server
+        };
+
+        // order in which the channels show up in the channel selector
+        private static readonly IReadOnlyList<ChatChannel> ChannelSelectorOrder = new List<ChatChannel>
+        {
+            ChatChannel.Local, ChatChannel.Emotes, ChatChannel.Radio, ChatChannel.OOC, ChatChannel.Dead, ChatChannel.AdminChat
+        };
+
+        private const float FilterPopupWidth = 110;
+        private const int DragMarginSize = 7;
+        private const float MinHeight = 128;
+        private const int MinWidth = 200;
+        private const int MinDistanceFromBottom = 255;
+        private const int MinLeft = 500;
+
+        /// <summary>
+        /// Will be Unspecified if set to Console
+        /// </summary>
+        public ChatChannel _selectedChannel;
```

Comment:
There are a bunch of minor naming violations like this. Just replace with PascalCase.